src/main/java/org/jilt/utils/AnnotationUtils.java

```diff
@@ -0,0 +1,42 @@
+package org.jilt.utils;
+
+import javax.lang.model.element.AnnotationMirror;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import java.util.EnumSet;
+import java.util.Set;
+
+public final class AnnotationUtils {
+    private static final Set<ElementType> NON_PARAMETER_ANNOTATIONS = EnumSet.of(ElementType.TYPE, ElementType.FIELD,
+            ElementType.METHOD, ElementType.CONSTRUCTOR, ElementType.LOCAL_VARIABLE, ElementType.ANNOTATION_TYPE,
+            ElementType.PACKAGE, ElementType.TYPE_PARAMETER);
+
+    static {
+        try {
+            NON_PARAMETER_ANNOTATIONS.add((ElementType) ElementType.class.getField("RECORD_COMPONENT").get(null));
+            NON_PARAMETER_ANNOTATIONS.add((ElementType) ElementType.class.getField("MODULE").get(null));
+        } catch (NoSuchFieldException e) {
+            // Java version does not support RECORD_COMPONENT
+        } catch (IllegalAccessException e) {
+            // Java version does not support RECORD_COMPONENT
+        }
+    }
```

Comment:
I actually don't think this is the correct way to check whether an annotation can be placed on a parameter.

An annotation can be placed on a parameter if either it's not annotated with `@Target` at all (which means it can be applied to any element), or, if `@Target` is present, `@Target.value` contains either `ElementType.PARAMETER` or `ElementType.TYPE_USE`. All other `ElementType` values are immaterial.

So, you can safely get rid of `NON_PARAMETER_ANNOTATIONS`, and everything related to it.

Given how simple that check is, please get rid of `AnnotationUtils`, and move the logic into `AbstractBuilderGenerator`.

Reply:
I felt I was overcomplicating thing, and indeed I was! Thanks for your comments